### DISCUSSION TITLE
How can I start the job immediately and after run on a interval

### DISCUSSION THREAD

**Discussion by tungrix** [kelektiv/node-cron#776] [Q&A] [answered]
Now, if I run the below script, the function will run after 5 seconds, instead of calling the function once before creating the Cronjob, is there any other way to run the function at the beginning first and then run every 5 seconds?
```javascript
import { CronJob } from 'cron';

const job = new CronJob(
	'*/5 * * * * *', // cronTime
	function () {
		console.log('You will see this message every 5 second');
	}, // onTick
	null, // onComplete
	true, // start
	'America/Los_Angeles' // timeZone
);

```

**Comment by sheerlox** [comment, marked as answer]
As specified in the [API documentation](https://github.com/kelektiv/node-cron#cronjob-class), you can use the `runOnInit` option to meet that requirement:

```javascript
import { CronJob } from 'cron';

const job = new CronJob(
  '*/5 * * * * *', // cronTime
  function () {
    console.log('You will see this message every 5 second');
  }, // onTick
  null, // onComplete
  true, // start
  'America/Los_Angeles', // timeZone
  null, // context
  true // runOnInit
);
```

On a side note, because it is more explicit and does not care about the order of options, I would recommend using the `CronJob.from()` static method to declare your jobs:
```javascript
import { CronJob } from 'cron';

const job = CronJob.from({
  cronTime: '*/5 * * * * *',
  onTick: function () {
    console.log('You will see this message every 5 second');
  },
  start: true,
  timeZone: 'America/Los_Angeles',
  runOnInit: true
});
```

**Reply by tungrix** [reply]
```javascript
import { CronJob } from "cron";

const creditRateJob = new CronJob.from({
  cronTime: "*/5 * * * * *",
  onTick: updateCreditRate,
  start: true,
  timeZone: "America/Los_Ang/eles",
  runOnInit: true,
});
```
When I ran the above code, I got the following error:
```
const creditRateJob = new CronJob.from({
                       ^
TypeError: CronJob.from is not a constructor
```
I think the constructor method `new` needs to be removed. Thank you

**Reply by sheerlox** [reply]
Indeed I forgot to remove it after copy-pasting, thanks for pointing this out, updated my answer!

**Comment by tungrix** [comment]
Thank you very much, especially for the side note, very helpful!
